### PR TITLE
Flexible expanded callback

### PR DESCRIPTION
**Pull request by sdementen**
allow a more flexible use of expanded_callback:
- no obligation to adapt the signature of a callback (by adding a **kwargs) when going from app.callback to app.expanded_callback 
- allow to specify the name of the specific extra parameters needed by the function (no need for **kwargs) to improve readability like in

```python
# current version
@app.expanded_callback(...)
def my_callback(..., **kwargs): ...

# new flexible version : no need to add **kwargs if not needed
@app.expanded_callback(...)
def my_callback(...): ...

# new flexible version : just get two of the new parameters (user and callback_context
@app.expanded_callback(...)
def my_callback(..., user, callback_context): ...

# new flexible version : get two of the new parameters and the rest in **kwargs
@app.expanded_callback(...)
def my_callback(..., user, callback_context, **kwargs): ...

```

### CODE REVIEW

**Comment by sdementen** [comment]
@GibbsConsulting I wonder even if, with the introspection of the callback signature, the distinction between expanded_callback and callback is necessary anymore...

Maybe a simple option at the DjangoDash(..., use_dash_dispatch=True|False) with a default to False would be cleaner.

**Comment by sdementen** [comment]
fix #306

**Comment by GibbsConsulting** [comment]
@sdementen yes, simplifying the callback/extended_callback approach is a great idea. Thanks for these contributions, most appreciated.

**Comment by sdementen** [comment]
Would using always the django dispatching be an option? Or do you think it is worth keeping the dash direct dispatch?
If the first option, then we could make all callback automatically expanded_callbcks with the trick of specifying in the callback function the name of the extra pdp arguments that pdp would inject automatically (the logic in this PR)

**Comment by sdementen** [comment]
I have integrated the expanded_callback functionality in the standard callback (now both functions are the same). It should be backward compatible with django dash app that are only using the normal arguments

**Comment by GibbsConsulting** [comment]
Other than any backwards compatibility issue, I cant see a reason for keeping the dash direct dispatch option. However, other than a possible extra overhead, which itself could not be large, I am unable to construct an immediate example of something that wouldn't work as a result of this change.

File: django_plotly_dash/dash_wrapper.py [hunk 1]
```diff
@@ -287,26 +284,60 @@ def form_dash_instance(self, replacements=None, ndid=None, base_pathname=None):
...
+        elif inspect.Parameter.VAR_POSITIONAL in parameter_types:
+            # there is a *args, assume all parameters afterwards (KEYWORD_ONLY) are to be injected
+            # some of these parameters may not be expanded arguments but that is ok
+            expanded = parameter_types.get(inspect.Parameter.KEYWORD_ONLY, [])
+        else:
+            # there is no **kwargs, filter argMap to take only the keyword arguments
+            expanded = parameter_types.get(inspect.Parameter.POSITIONAL_OR_KEYWORD, [])[
+                       n_dash_parameters:] + parameter_types.get(inspect.Parameter.KEYWORD_ONLY, [])
+
+        print(func, parameter_types, "->",expanded)
```
Comment (GibbsConsulting) [inline django_plotly_dash/dash_wrapper.py:310]: This isn't the only one I'm sure, but we should probably start removing these print statements from the codebase or at least change them to use of logging.

**Review by GibbsConsulting — APPROVED** [review]

File: django_plotly_dash/dash_wrapper.py (hunk 1 above)
Comment (sdementen) [reply django_plotly_dash/dash_wrapper.py]: I have removed this + two others in an other file

Comment (GibbsConsulting) [reply django_plotly_dash/dash_wrapper.py]: Thats great, thanks!

**Comment by zvi-quantivly** [comment]
> I am unable to construct an immediate example of something that wouldn't work as a result of this change.
Just for completeness’ sake, mentioning here #484, which, IIUC, is caused by this.

**Pull request merged**